src/screens/KeyBasedAssetTransfer/KeyBasedAssetTransferStatus.js

```diff
@@ -0,0 +1,174 @@
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+import React, { useEffect } from 'react';
+import { connect } from 'react-redux';
+import isEmpty from 'lodash.isempty';
+import styled from 'styled-components/native';
+import type { NavigationScreenProp } from 'react-navigation';
+
+// constants
+import { TRANSACTION_EVENT } from 'constants/historyConstants';
+import { COLLECTIBLES } from 'constants/assetsConstants';
+import { COLLECTIBLE_TRANSACTION } from 'constants/collectiblesConstants';
+
+// actions
+import {
+  checkKeyBasedAssetTransferTransactionsAction,
+  setAndStoreKeyBasedAssetsToTransferAction,
+} from 'actions/keyBasedAssetTransferActions';
+
+// components
+import ContainerWithHeader from 'components/Layout/ContainerWithHeader';
+import { Footer, ScrollWrapper, Wrapper } from 'components/Layout';
+import Button from 'components/Button';
+import ActivityFeed from 'components/ActivityFeed';
+import Animation from 'components/Animation';
+import { MediumText } from 'components/Typography';
+
+// utils
+import { mapTransactionsHistory } from 'utils/feedData';
+import { buildHistoryTransaction } from 'utils/history';
+import { parseTokenBigNumberAmount } from 'utils/common';
+import { fontStyles, spacing } from 'utils/variables';
+import t from 'translations/translate';
+
+// types
+import type { KeyBasedAssetTransfer } from 'models/Asset';
+import type { Dispatch, RootReducerState } from 'reducers/rootReducer';
+import type { Accounts } from 'models/Account';
+
+
+type Props = {
+  navigation: NavigationScreenProp<*>,
+  resetKeyBasedAssetsTransfer: () => void,
+  checkKeyBasedAssetTransferTransactions: () => void,
+  keyBasedAssetsTransfer: KeyBasedAssetTransfer[],
+  accounts: Accounts,
+};
+
+const Title = styled(MediumText)`
+  ${fontStyles.large};
+  text-align: center;
+  margin-bottom: ${spacing.small}px;
+`;
+
+const animationSuccess = require('assets/animations/transactionSentConfirmationAnimation.json');
+
+const KeyBasedAssetTransferStatus = ({
+  navigation,
+  checkKeyBasedAssetTransferTransactions,
+  keyBasedAssetsTransfer,
+  accounts,
+  resetKeyBasedAssetsTransfer,
+}: Props) => {
+  useEffect(() => { checkKeyBasedAssetTransferTransactions(); }, []);
+
+  const assetTransferTransactions = keyBasedAssetsTransfer.map(({
+    signedTransaction,
+    transactionHash,
+    amount,
+    status,
+    assetData,
+  }) => ({
+    ...buildHistoryTransaction({
+      from: signedTransaction?.from || '',
+      to: signedTransaction?.to || '',
+      hash: transactionHash || '',
```

Comment:
I'm assuming this ends up being rendered by the UI so hence the `''` - which i can understand

Reply:
Yes, this is added to not crash the UI if somehow values didn't make up to here, however, we can logs elsewhere for such cases.